src/model/createModel.ts

```diff
@@ -64,14 +69,18 @@ export function createModel<
         return properties
       }
 
-      if (
-        typeof initialValue === 'string' ||
-        typeof initialValue === 'number' ||
-        typeof initialValue === 'boolean' ||
-        // @ts-ignore
-        initialValue?.constructor.name === 'Date' ||
-        Array.isArray(initialValue)
-      ) {
+      if (propertyDefinition instanceof NullableProperty) {
+        set(
+          properties,
+          propertyName,
+          initialValue === null || isModelValueType(initialValue)
```

Comment:
Do you mind extracting this value expression into a variable? It'd read much nicer.

```ts
const value = initialValue === null || isModelValueType(initialValue) ? initialValue : propertyDefinition.getValue()
set(properties, propertyName, value)
```